src/librbd/cache/pwl/Types.h

```diff
@@ -278,32 +283,30 @@ struct WriteLogCacheEntry {
 };
 
 struct WriteLogPoolRoot {
-  #ifdef WITH_RBD_RWL
-  union {
-    struct {
-      uint8_t layout_version;    /* Version of this structure (RWL_POOL_VERSION) */
-    };
-    uint64_t _u64;
-  } header;
-  TOID(struct WriteLogCacheEntry) log_entries;   /* contiguous array of log entries */
-  #endif
-  #ifdef WITH_RBD_SSD_CACHE
-  uint64_t layout_version = 0;
-  uint64_t cur_sync_gen = 0;
-  #endif
+  uint64_t layout_version;
   uint64_t pool_size;
   uint64_t flushed_sync_gen;     /* All writing entries with this or a lower
                                   * sync gen number are flushed. */
   uint32_t block_size;           /* block size */
-  uint32_t num_log_entries;
+  uint64_t num_log_entries;
   uint64_t first_free_entry;     /* Entry following the newest valid entry */
   uint64_t first_valid_entry;    /* Index of the oldest valid entry in the log */
 
+  union {
+    /* RWL */
+    struct {
+      PMEMoid log_entries;      /* contiguous array of log entries */
+    };
+    /* SSD */
+    struct {
+      uint64_t unused;
```

Comment:
what is the use-case for `unused` variable here?

Reply:
> what is the use-case for `unused` variable here?

It is unused, but it won't take up any extra space because it is in `union`. I just want to keep the SSD structure so that we can conveniently extend the fields in the future.